common/authorization/interceptor.go

```diff
@@ -45,42 +49,77 @@ func (a *interceptor) Interceptor(
 	handler grpc.UnaryHandler,
 ) (interface{}, error) {
 
-	var namespace string
-	requestWithNamespace, ok := req.(requestWithNamespace)
-	if ok {
-		namespace = requestWithNamespace.GetNamespace()
-	}
+	var claims *Claims
 
-	apiName := info.FullMethod
-	index := strings.LastIndex(apiName, "/")
-	if index > -1 {
-		apiName = apiName[index+1:]
+	if a.claimMapper != nil && a.authorizer != nil {
+		var tlsSubject *pkix.Name
+		var authHeaders []string
+
+		if md, ok := metadata.FromIncomingContext(ctx); ok {
+			authHeaders = md["authorization"]
+		}
+		if p, ok := peer.FromContext(ctx); ok {
+			if tlsAuth, ok := p.AuthInfo.(credentials.TLSInfo); ok {
+				if len(tlsAuth.State.VerifiedChains) > 0 && len(tlsAuth.State.VerifiedChains[0]) > 0 {
+					tlsSubject = &tlsAuth.State.VerifiedChains[0][0].Subject
+				}
+			}
+		}
+		// Add auth into to ctx only if there's some auth info
+		if tlsSubject != nil || len(authHeaders) > 0 {
+			var authHeader string
+			if len(authHeaders) > 0 {
```

Comment:
what do we do with the "rest of the headers" (like if there are more than 1), or do we pretty much expect this to either be empty or 1?

Reply:
My understanding is that even though it is technically possible to provide multiple tokens in an "Authorization" header, that is primarily used when multiple authorization options are supported (e.g. OAuth and basic username/password), and most apps use a single token.

When somebody is making a call to a temporal API, I think we can expect a single token from them. We can extend this logic in the future if we hear from customers otherwise.